apheleia-core.el

```diff
@@ -768,7 +768,11 @@ cmd is to be run."
                     "node_modules"
                     project-dir)))))
             (when (file-executable-p binary)
-              (setcar command binary)))))
+              (setcar command binary))))
+        (when (locate-dominating-file default-directory ".pnp.cjs")
+          (if-let (yarn-executable (executable-find "yarn"))
+              (push yarn-executable command)
+            (message "apheleia: .pnp.cjs found but yarn is not installed"))))
```

Comment:
So, as someone who hasn't used Yarn PNP yet, is the expected usage that you run `yarn prettier` in order to invoke the correct version of Prettier? And `npx prettier` does not work with PNP projects?

Reply:
Yes, PNP mode does not create a node_modules/ dir, and `yarn foo` is how you run your locally installed executable `foo` using all the locally installed dependencies.

Reply:
If the binary isn't managed by your project, does Yarn fall back to invoking from the rest of `$PATH`? If not, we should add a check to ensure that the binary exists before prepending `yarn`.

Reply:
I think that's handled by line 773? 
```elisp
(if-let (yarn-executable (executable-find "yarn"))` ?
```
I cribbed the call to `executable-find` from the code that handles npx/node-modules, so I might be misunderstanding what it does..?

Reply:
No, I don't mean check if the `yarn` binary exists, I mean check if the binary the user wants to invoke is installed as part of the current project. If you're in a PNP project that does not have Prettier installed as a dependency, then Apheleia should fall back to invoking a system-global Prettier instead.

Reply:
Ah!  I see now.

I can think of three ways to address this:

1. Parse the project's package.json file to see if prettier is locally installed.  I'd want to use elisp's json-parse-string for this, which won't work if emacs hasn't been compiled with json support.
2. Use `(shell-exec-to-string (concat yarn-executabnle " info prettier"))`.  This will reintroduce the delay @zarybnicky [noted](https://github.com/radian-software/apheleia/pull/200#issuecomment-1656687846) with the yarn version check.
3. Check for the presence of `.yarn/sdks/prettier/`.  This will not work unless the user has manually installed the base sdks via `yarn dlx @yarnpkg/sdks base`

Thoughts?  Other ideas?

Reply:
I think there is a faster and more reliable way to get the subset of information we are looking for. For example, here is something close to finding the path of prettier:

```
% time node .pnp.js prettier apheleia
[null,"/home/raxod502/.cache/yarn/v6/npm-prettier-3.0.3-432a51f7ba422d1469096c0fdc28e235db8f9643-integrity/node_modules/prettier/index.cjs"]
node .pnp.js prettier apheleia  0.02s user 0.01s system 106% cpu 0.031 total
```

This needs a bit more work to figure out. I will continue looking at this and update here.